api/lib/domain/services/schooling-registrations-xml-service.js

```diff
@@ -37,86 +37,84 @@ async function extractSchoolingRegistrationsInformationFromSIECLE(path, organiza
-
-    streamerToParseOrganizationUAI.once('match', (xmlNode) => {
-      xml2js.parseString(xmlNode, (err, nodeData) => {
-        if (err) return reject(err);
-        if (nodeData.PARAMETRES) {
-          const UAIFromSIECLE = _getValueFromParsedElement(nodeData.PARAMETRES.UAJ);
-          resolve(UAIFromSIECLE);
-        }
-      });
-    });
+function _UAIextractor(saxParser, resolve, reject) {
+  const streamerToParseOrganizationUAI = new saxPath.SaXPath(saxParser, NODE_ORGANIZATION_UAI);
 
-    saxParser.on('end', () => {
-      reject(new FileValidationError(UAI_SIECLE_FILE_NOT_MATCH_ORGANIZATION_UAI));
+  streamerToParseOrganizationUAI.once('match', (xmlNode) => {
+    xml2js.parseString(xmlNode, (err, nodeData) => {
+      if (err) return reject(err);
+      if (nodeData.PARAMETRES) {
+        const UAIFromSIECLE = _getValueFromParsedElement(nodeData.PARAMETRES.UAJ);
+        resolve(UAIFromSIECLE);
+      }
     });
+  });
 
-    siecleFileStream.pipe(saxParser);
+  saxParser.on('end', () => {
+    reject(new FileValidationError(UAI_SIECLE_FILE_NOT_MATCH_ORGANIZATION_UAI));
   });
 }
 
-function _extractStudentRegistrationsFromStream(siecleFileStream) {
-  return new Promise(function(resolve, reject_) {
-    const saxParser = sax.createStream(true);
-    saxParser.on('error', () => {
-      reject(new FileValidationError(NO_STUDENTS_IMPORTED_FROM_INVALID_FILE));
-    });
+function _registrationExtractor(saxParser, resolve, reject) {
+  const mapSchoolingRegistrationsByStudentId = new Map();
+  const nationalStudentIds = [];
 
-    const reject = (e) => {
-      saxParser.removeAllListeners();
-      return reject_(e);
-    };
-
-    const mapSchoolingRegistrationsByStudentId = new Map();
-    const nationalStudentIds = [];
-
-    const streamerToParseSchoolingRegistrations = new saxPath.SaXPath(saxParser, NODES_SCHOOLING_REGISTRATIONS);
-    streamerToParseSchoolingRegistrations.on('match', (xmlNode) => {
-      if (_isSchoolingRegistrationNode(xmlNode)) {
-        xml2js.parseString(xmlNode, (err, nodeData) => {
-          try {
-            if (err) throw err;
-            processStudentsNodes(mapSchoolingRegistrationsByStudentId, nodeData, nationalStudentIds);
-            processStudentsStructureNodes(mapSchoolingRegistrationsByStudentId, nodeData);
-          } catch (err) {
-            reject(err);
-          }
-        });
-      }
-    });
+  const streamerToParseSchoolingRegistrations = new saxPath.SaXPath(saxParser, NODES_SCHOOLING_REGISTRATIONS);
 
-    siecleFileStream.pipe(saxParser);
+  streamerToParseSchoolingRegistrations.on('match', (xmlNode) => {
+    if (_isSchoolingRegistrationNode(xmlNode)) {
+      xml2js.parseString(xmlNode, (err, nodeData) => {
+        try {
+          if (err) throw err;
+          processStudentsNodes(mapSchoolingRegistrationsByStudentId, nodeData, nationalStudentIds);
+          processStudentsStructureNodes(mapSchoolingRegistrationsByStudentId, nodeData);
```

Comment:
Ces deux fonctions semblent locales au fichier.

```suggestion
          _processStudentsNodes(mapSchoolingRegistrationsByStudentId, nodeData, nationalStudentIds);
          _processStudentsStructureNodes(mapSchoolingRegistrationsByStudentId, nodeData);
```

Reply:
Done